tests/common/test_host_basics.hpp

```diff
@@ -425,9 +718,124 @@ TYPED_TEST_P(test_host_basics_matrix, matrix_3x3) {
     ASSERT_NEAR(algebra::getter::element(m33_inv, 2, 2), -10.f / 20.f,
                 this->m_isclose);
   }
+
+  {
+    typename TypeParam::template matrix<N, N> m1;
```

Comment:
Is this the same test code as below? Can this be written as a generic function or is that not possible?

Reply:
Great catch; I've deduplicated the testing code.